pandas/core/generic.py

```diff
@@ -6322,7 +6322,10 @@ def convert_dtypes(
                 )
                 for col_name, col in self.items()
             ]
-            return concat(results, axis=1, copy=False)
+            if len(results) > 0:
+                return concat(results, axis=1, copy=False)
+            else:
+                return self
```

Comment:
Since it's empty it might not matter that much, but to be sure let's return `self.copy()` (the other path through concat will also always return a new object)